### PR TITLE
[Bug] [source elasticsearch] The 'source' field is mandatory (#7420)

### PR DESCRIPTION
**Pull request by zhdech**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ support version >= 2.x and <= 8.x.
 | username                | string  | no       | -                 |
 | password                | string  | no       | -                 |
 | index                   | string  | yes      | -                 |
-| source                  | array   | no       | -                 |
+| source                  | array   | yes      | -                 |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Elasticsearch.md]: The `source` is not required. We can use `schema` too. Please refer https://github.com/apache/seatunnel/pull/7421/files#diff-8eef53f88ae098e1477f67d8053f946d3267578ec8e68fd472d22c298864f876R128. Could you please update the documentation to make sure users are aware of this behavior?

Comment (zhdech) [reply docs/en/connector-v2/source/Elasticsearch.md]: Is this description feasible?

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 2]
```diff
@@ -54,11 +55,15 @@ x-pack password.
...
+### schema [config]
+
+The schema config in ElasticSearch sink is deprecated, please use source config instead! Choose between schema and source, it is recommended to use source
+
 ### source [array]
 
 The fields of index.
 You can get the document id by specifying the field `_id`.If sink _id to other index,you need specify an alias for _id due to the Elasticsearch limit.
-If you don't config source, it is automatically retrieved from the mapping of the index.
+If you don't config source, it is automatically retrieved from the mapping of the index.Choose between schema and source, it is recommended to use source
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Elasticsearch.md]:
```suggestion
If you don't config source, it is automatically retrieved from the mapping of the index.
```

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 3]
```diff
@@ -54,11 +55,15 @@ x-pack password.
 
 Elasticsearch index name, support * fuzzy matching.
 
+### schema [config]
+
+The schema config in ElasticSearch sink is deprecated, please use source config instead! Choose between schema and source, it is recommended to use source
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Elasticsearch.md]:
```suggestion
The structure of the data, including field names and field types. Reference to [Schema-Feature](../../concept/schema-feature.md)
The schema config in ElasticSearch sink is deprecated, please use source config instead.
```

**Comment by zhdech** [comment]
`If you don't config source, it is automatically retrieved from the mapping of the index.`
In practical applications, if 'source' is not configured, it cannot be automatically obtained and an error will be reported directly. Please refer to the following bug for details. May I ask how Mr. Fan's situation should be resolved?
https://github.com/apache/seatunnel/issues/7420

**Comment by Hisoka-X** [comment]
> `If you don't config source, it is automatically retrieved from the mapping of the index.` In practical applications, if 'source' is not configured, it cannot be automatically obtained and an error will be reported directly. Please refer to the following bug for details. May I ask how Mr. Fan's situation should be resolved? #7420

I believe your connector version isn't 2.3.6. It already fixed in https://github.com/apache/seatunnel/pull/6181 which released in 2.3.4.

**Pull request closed without merging**